CONTRIBUTING.md

```diff
@@ -78,21 +78,44 @@ See <https://docs.basedpyright.com/> for IDE integration and other info.
 
 ### mypy
 
-Mypy requires a bit more effort to run, since it doesn't play well with
-editable dependencies, and it doesn't seem to recognize NumType's `numpy-stubs`
-if `numpy` is also installed. You can get around that by running:
+Mypy is used for type-checking and specific tests. It requires more effort to run
+correctly because it doesn't handle editable dependencies well and needs isolation
+from the main numpy installation to recognize NumType's numpy-stubs.
+
+#### General Type Checking
```

Comment:
It's not a bad idea to discuss the type-testing and type-checking separately; they serve different purposes after all. 
But mypy (and basedpyright) can do both using one command. So I don't see any benefit for splitting up the commands. It'll make it more complicated that it needs to be.